attestation-agent/attestation-agent/src/bin/grpc-aa/server.rs

```diff
@@ -162,6 +163,31 @@ impl AttestationAgentService for AA {
 
         Result::Ok(Response::new(reply))
     }
+
+    async fn get_tee_type(
+        &self,
+        _request: Request<GetTeeTypeRequest>,
+    ) -> Result<Response<GetTeeTypeResponse>, Status> {
+        let mut attestation_agent = self.inner.lock().await;
+
+        debug!("AA (grpc): get tee type ...");
+
+        let tee = attestation_agent.get_tee_type();
+
+        let tee = serde_json::to_string(&tee)
+            .map_err(|e| {
+                error!("AA (ttrpc): get tee type failed:\n {e:?}");
+                Status::internal(format!("[ERROR:{AGENT_NAME}] AA get tee type failed"))
+            })?
+            .trim_end_matches('"')
+            .trim_start_matches('"')
+            .to_string();
```

Comment:
This json hack is a bit weird. Is there no way to deserialize as a string?

Reply:
The serialization comes from the enum `Tee` from `kbs-types`. In rust a `enum` will be serialized into forms like `"xxx"` rather than `xxx`. Thus we need to manually trim the `"` at both starting and end.